cmd/mattermost/commands/config.go

```diff
@@ -318,6 +326,88 @@ func UpdateMap(configMap map[string]interface{}, configSettings []string, newVal
 	}
 }
 
+func configResetCmdF(command *cobra.Command, args []string) error {
+	app, err := InitDBCommandContextCobra(command)
+	if err != nil {
+		return err
+	}
+
+	defer app.Shutdown()
+
+	defaultConfig := &model.Config{}
+	defaultConfig.SetDefaults()
+
+	if len(args) == 0 {
+		var confirmResetAll string
+		CommandPrettyPrintln("Are you sure you want to reset all the configuration settings?(YES/NO): ")
+		fmt.Scanln(&confirmResetAll)
+		if confirmResetAll == "YES" {
+			app.SaveConfig(defaultConfig, false)
+
+		}
+	} else {
+		tempConfig := app.Config()
+		tempConfigMap := configToMap(*tempConfig)
+		defaultConfigMap := configToMap(*defaultConfig)
+		for _, arg := range args {
+			err := changeMap(tempConfigMap, defaultConfigMap, strings.Split(arg, "."))
+			if err != nil {
+				return errors.Wrap(err, "Failed to reset config")
+			}
+		}
+		bs, err := json.Marshal(tempConfigMap)
+		if err != nil {
+			fmt.Printf("Error while marshalling map to json %s\n", err)
+			os.Exit(1)
+		}
+		err = json.Unmarshal(bs, tempConfig)
+		if err != nil {
+			fmt.Printf("Error while unmarshalling json to struct %s\n", err)
+			os.Exit(1)
+		}
+		app.SaveConfig(tempConfig, false)
+	}
+
+	newConfig := app.Config()
+	if err := newConfig.IsValid(); err != nil {
```

Comment:
The IsValid is called always as part of the `SaveConfig` function (actually, not exactly there, but if you deep in the different implementations of config storage, before each save is called the validation).